From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: bejiihiu <bejiihiu@icloud.com>
Date: Wed, 17 Jun 2026 23:50:13 +0500
Subject: [PATCH] Rebrand Folia to FuckTick


diff --git a/src/main/java/com/destroystokyo/paper/Metrics.java b/src/main/java/com/destroystokyo/paper/Metrics.java
index 08745a35b1ce8cc56d970472088de3cd17b9e43a..915b27b7dc0e7a4afa0981c5a3d90c0d1a5c79d9 100644
--- a/src/main/java/com/destroystokyo/paper/Metrics.java
+++ b/src/main/java/com/destroystokyo/paper/Metrics.java
@@ -593,7 +593,7 @@ public class Metrics {
             boolean logFailedRequests = config.getBoolean("logFailedRequests", false);
             // Only start Metrics, if it's enabled in the config
             if (config.getBoolean("enabled", true)) {
-                Metrics metrics = new Metrics("Folia", serverUUID, logFailedRequests, Bukkit.getLogger()); // Folia - we have our own bstats page
+                Metrics metrics = new Metrics("FuckTick", serverUUID, logFailedRequests, Bukkit.getLogger()); // Folia - we have our own bstats page
 
                 metrics.addCustomChart(new Metrics.SimplePie("minecraft_version", () -> {
                     String minecraftVersion = Bukkit.getVersion();
@@ -607,11 +607,11 @@ public class Metrics {
                 final String implVersion = org.bukkit.craftbukkit.Main.class.getPackage().getImplementationVersion();
                 if (implVersion != null) {
                     final String buildOrHash = implVersion.substring(implVersion.lastIndexOf('-') + 1);
-                    paperVersion = "git-Folia-%s-%s".formatted(Bukkit.getServer().getMinecraftVersion(), buildOrHash); // Folia - we have our own bstats page
+                    paperVersion = "git-FuckTick-%s-%s".formatted(Bukkit.getServer().getMinecraftVersion(), buildOrHash); // Folia - we have our own bstats page
                 } else {
                     paperVersion = "unknown";
                 }
-                metrics.addCustomChart(new Metrics.SimplePie("folia_version", () -> paperVersion)); // Folia - we have our own bstats page
+                metrics.addCustomChart(new Metrics.SimplePie("fucktick_version", () -> paperVersion)); // Folia - we have our own bstats page
 
                 metrics.addCustomChart(new Metrics.DrilldownPie("java_version", () -> {
                     Map<String, Map<String, Integer>> map = new HashMap<>();
diff --git a/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java b/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
index 4d2936f6818c38e2844cba78ffad339e93ff29ab..377774484d78cf3b9ecabfd9c88db7eb0d44c8bb 100644
--- a/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
+++ b/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
@@ -40,8 +40,8 @@ public class PaperVersionFetcher implements VersionFetcher {
     private static final ComponentLogger COMPONENT_LOGGER = ComponentLogger.logger(LogManager.getRootLogger().getName());
     private static final int DISTANCE_ERROR = -1;
     private static final int DISTANCE_UNKNOWN = -2;
-    private static final String DOWNLOAD_PAGE = "https://papermc.io/downloads/folia"; // Folia
-    private static final String REPOSITORY = "PaperMC/Folia"; // Folia
+    private static final String DOWNLOAD_PAGE = "https://papermc.io/downloads/fucktick"; // Folia
+    private static final String REPOSITORY = "PaperMC/fucktick"; // Folia
     private static final ServerBuildInfo BUILD_INFO = ServerBuildInfo.buildInfo();
     private static final String USER_AGENT = BUILD_INFO.brandName() + "/" + BUILD_INFO.asString(VERSION_SIMPLE) + " (https://papermc.io)";
     private static final Gson GSON = new Gson();
@@ -139,7 +139,7 @@ public class PaperVersionFetcher implements VersionFetcher {
         final String currentVersion = PaperVersionFetcher.BUILD_INFO.minecraftVersionId();
 
         try {
-            final URL versionsUrl = URI.create("https://fill.papermc.io/v3/projects/folia").toURL(); // Folia
+            final URL versionsUrl = URI.create("https://fill.papermc.io/v3/projects/fucktick").toURL(); // Folia
             final HttpURLConnection connection = (HttpURLConnection) versionsUrl.openConnection();
             connection.setConnectTimeout(5000);
             connection.setReadTimeout(5000);
@@ -161,7 +161,7 @@ public class PaperVersionFetcher implements VersionFetcher {
                     }
 
                     try {
-                        final URL buildsUrl = URI.create("https://fill.papermc.io/v3/projects/folia/versions/" + latestVersion + "/builds/latest").toURL(); // Folia
+                        final URL buildsUrl = URI.create("https://fill.papermc.io/v3/projects/fucktick/versions/" + latestVersion + "/builds/latest").toURL(); // Folia
                         final HttpURLConnection connection2 = (HttpURLConnection) buildsUrl.openConnection();
                         connection2.setConnectTimeout(5000);
                         connection2.setReadTimeout(5000);
@@ -194,7 +194,7 @@ public class PaperVersionFetcher implements VersionFetcher {
 
     private static int fetchDistanceFromSiteApi(final int jenkinsBuild) {
         try {
-            final URL buildsUrl = URI.create("https://fill.papermc.io/v3/projects/folia/versions/" + PaperVersionFetcher.BUILD_INFO.minecraftVersionId() + "/builds").toURL(); // Folia
+            final URL buildsUrl = URI.create("https://fill.papermc.io/v3/projects/fucktick/versions/" + PaperVersionFetcher.BUILD_INFO.minecraftVersionId() + "/builds").toURL(); // Folia
             final HttpURLConnection connection = (HttpURLConnection) buildsUrl.openConnection();
             connection.setConnectTimeout(5000);
             connection.setReadTimeout(5000);
diff --git a/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java b/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
index 10a0be7a4db1a51579d113d279af7a9effe7f438..99067f64bac02cf1ebcecb2e32688f92eab6c196 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
@@ -44,7 +44,7 @@ public record ServerBuildInfoImpl(
                 .map(Key::key)
                 .orElse(Key.key("papermc", "folia")), // Folia
             getManifestAttribute(manifest, ATTRIBUTE_BRAND_NAME)
-                .orElse("Folia"), // Folia
+                .orElse("FuckTick"), // Folia
             SharedConstants.getCurrentVersion().id(),
             SharedConstants.getCurrentVersion().name(),
             getManifestAttribute(manifest, ATTRIBUTE_BUILD_NUMBER)
