From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: bejiihiu <bungeepithub@gmail.com>
Date: Fri, 19 Jun 2026 14:37:13 +0500
Subject: [PATCH] Shutdown WatchCat with compute broker


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index 2db84ba63df8dd646226fb4c5422e2b640466d03..491f9d1664d859b084a4ed728f021a3f9c131224 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1120,7 +1120,8 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
 
         LOGGER.info("Stopping server");
         Commands.COMMAND_SENDING_POOL.shutdownNow(); // Paper - Perf: Async command map building; Shutdown and don't bother finishing
-        // FuckTick Start - остановка compute broker
+        // FuckTick Start - остановка WatchCat и compute broker
+        io.papermc.paper.fucktick.watchcat.WatchCat.shutdownIfInitialized();
         io.papermc.paper.fucktick.compute.FuckTickComputeBroker.shutdownIfInitialized();
         // FuckTick End
         // CraftBukkit start
